jobclient/python/cookclient/__init__.py

```diff
@@ -279,6 +341,47 @@ def close(self):
         if self.__session is not None:
             self.__session.close()
 
+    @staticmethod
+    def _apply_jobspec_defaults(jobspec: dict):
+        """Apply default values to a jobspec.
+
+        This function will add default values to a jobspec if no value is
+        provided. The provided jobspec will be modified in-place.
+        """
+        if 'uuid' not in jobspec:
+            jobspec['uuid'] = str(util.make_temporal_uuid())
+        if 'cpus' not in jobspec:
+            jobspec['cpus'] = 1.0
+        if 'mem' not in jobspec:
+            jobspec['mem'] = 128.0
+        if 'max-retries' not in jobspec:
+            jobspec['max-retries'] = 1
+        if 'max-runtime' not in jobspec:
+            jobspec['max-runtime'] = timedelta(days=1)
+        if 'name' not in jobspec:
+            jobspec['name'] = f'{getpass.getuser()}-job'
+        if 'application' not in jobspec:
+            jobspec['application'] = _CLIENT_APP
+
+    @staticmethod
+    def _convert_jobspec(jobspec: dict):
+        """Convert a Python jobspec into a JSON jobspec.
+
+        This function will convert the higher-level Python types used in job
+        submissions into their JSON primitive counterparts (e.g., timedelta is
+        converted into the number of milliseconds).
+
+        The provided jobspec is modified in-place.
```

Comment:
Modifying in place seems bad to me. E.g, calling this twice would be a problem. Don't like the refactor. Would suggest migrate this logic into submit().

Reply:
@hdrodz97

Reply:
Fixed them, they no longer modify in-place

The logic used to be in `submit()`, but these functions were added to make `submit_all()` a little bit cleaner